code/modules/organs/organ_external.dm

```diff
@@ -1382,6 +1382,20 @@ obj/item/weapon/organ/head/New(loc, mob/living/carbon/human/H)
 	brainmob.stat = 2
 	brainmob.death()
 
+	if(brainmob.mind && brainmob.mind.special_role == "highlander")
+		if(H.lastattacker && istype(H.lastattacker, /mob/living/carbon/human))
+			var/mob/living/carbon/human/L = H.lastattacker
+			if(L.mind && L.mind.special_role == "highlander")
+				L.revive(0)
+				to_chat(L, "<span class='notice'>You absorb \the [brainmob]'s power!</span>")
+				var/turf/T1 = get_turf(H)
+				spawn()
+					for(var/i = 0;i < 10;i++)
```

Comment:
You should make this into a proc, since you copy pasted it from soulstone code (I assume).